src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -43,6 +43,16 @@ public class HttpDocumentRetriever : IDocumentRetriever
         private HttpClient _httpClient;
         private static readonly HttpClient _defaultHttpClient = new HttpClient();
 
+        /// <summary>
+        /// Gets or sets whether additional headers are added to a <see cref="HttpRequestMessage"/> headers.
+        /// </summary>
+        public static bool AllowAdditionalHeaderData { get; set; } = true;
```

Comment:
ASP.NET users always ask for an easy opt-out of IM telemetry.
https://github.com/dotnet/aspnetcore/commit/32dd435c6e02267afddbfc715c7b92439f21d8cd
It looks like we'll need to update this line:
https://github.com/dotnet/aspnetcore/blob/fd75fcdbcd668a1dbf0a3da8f2ee8685850dab5d/src/Security/Authentication/OpenIdConnect/src/OpenIdConnectPostConfigureOptions.cs#L99-L100

Reply:
That looks right.